libs/expo/betterangels/src/lib/ui-components/RequestedProvidedServices/ServicesModal.tsx

```diff
@@ -0,0 +1,299 @@
+              error
+            );
+          }
+        }
+      }
+
+      for (const service of toCreateOtherServices) {
+        if (service.title) {
+          try {
+            await createService({
+              variables: {
+                data: {
+                  service: ServiceEnum.Other,
+                  serviceOther: service.title,
+                  noteId,
+                  serviceRequestType: type,
+                },
+              },
+            });
+          } catch (error) {
+            console.error(
+              `Error creating service with title ${service.title}:`,
+              error
+            );
+          }
+        }
+      }
+
+      refetch();
+      setIsModalVisible(false);
+    } catch (e) {
+      console.error('Error during service submission:', e);
+    }
+  };
+
+  const closeModal = () => {
+    const newInitialServices = initialServices
+      .filter((item) => item.service !== ServiceEnum.Other)
+      .map((service) => ({ id: service.id, enum: service.service }));
+    const initialServiceOthers = initialServices
+      .filter((item) => item.service === ServiceEnum.Other)
+      .map((service) => ({
+        id: service.id,
+        title: service.serviceOther || '',
+      }));
+    setIsModalVisible(false);
+    setServiceOthers(initialServiceOthers);
+    setServices(newInitialServices);
+  };
+
+  useEffect(() => {
+    const newInitialServices = initialServices
+      .filter((item) => item.service !== ServiceEnum.Other)
+      .map((service) => ({ id: service.id, enum: service.service }));
+    const initialServiceOthers = initialServices
+      .filter((item) => item.service === ServiceEnum.Other)
+      .map((service) => ({
+        id: service.id,
+        title: service.serviceOther || '',
+      }));
+
+    setServiceOthers(initialServiceOthers);
+    setServices(newInitialServices);
+  }, [initialServices]);
+
+  return (
+    <Modal
+      mt={Spacings.sm}
+      vertical
+      closeButton
+      closeModal={closeModal}
+      isModalVisible={isModalVisible}
+    >
+      <KeyboardAwareScrollView
+        style={{
+          flex: 1,
+          backgroundColor: Colors.WHITE,
+        }}
+        keyboardShouldPersistTaps="handled"
+      >
+        <ScrollView
+          contentContainerStyle={{
+            flexGrow: 1,
+            gap: Spacings.sm,
+            paddingBottom: Spacings.md,
+          }}
+          style={{ paddingHorizontal: Spacings.md }}
+        >
+          <TextBold> Services</TextBold>
+          <TextRegular mt="xxs" mb="sm">
+            Select the services to your client in this interaction.
+          </TextRegular>
+          {Services.map((service) => (
+            <View key={service.title}>
+              <TextBold mb="xs">{service.title}</TextBold>
+              {service.items.map((item, idx) => {
+                return (
+                  <ServiceCheckbox
+                    key={item.enum}
+                    services={services}
```

Comment:
I think this should be renamed a bit. By the time I get to this line I'm pretty confused and not sure what `services` means. Maybe renaming `Services` - the Services you map over - to `Categories` (categories have services) may help, assuming i'm following this correctly

Reply:
as I renamed all others I think we can keep "services" for the state

Reply:
So I'll leave it up to you, but here's a version to demonstrate what i mean.
I'm sure it's much more clear to you as you've super familiar with this code, but as I come in to read it's a bit of an effort to follow.

```javascript
{ServicesByCategory.map((category) => ( // we map over categories
  <View key={category.title}> // category has a title
    <TextBold mb="xs">{category.title}</TextBold>
    {category.services.map((service: ServiceEnum, idx) => { // category has services
      return (
        <ServiceCheckbox
          key={service}
          services={services} // which services?
          setServices={setServices} // which services are we setting?
          noteId={noteId} // not sure we need this
          service={service}
          idx={idx}
        />
      );
    })}
  </View>
))}
```